cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -274,79 +172,69 @@ angular.module(PKG.name + '.commons')
       }, 500);
     }
 
+    function closeNodePopover(node) {
+      var nodeInfo = nodePopovers[node.name];
+      if (nodePopoverTimeout) {
+        $timeout.cancel(nodePopoverTimeout);
+      }
+      if (nodeInfo && nodeInfo.popover) {
+        nodeInfo.popover.hide();
+        nodeInfo.popover.destroy();
+        nodeInfo.popover = null;
+      }
+    }
+
     vm.nodeMouseEnter = function (node) {
       if (!$scope.showMetrics || vm.scale >= SHOW_METRICS_THRESHOLD) { return; }
+
       var nodeInfo = nodePopovers[node.name];
 
-      nodeInfo.popover = $popover(nodeInfo.element, {
-        trigger: 'manual',
-        placement: 'auto right',
-        target: angular.element(nodeInfo.element[0]),
-        templateUrl: $scope.nodePopoverTemplate,
-        container: 'main',
-        scope: nodeInfo.scope
-      });
-      nodeInfo.popover.$promise
-        .then(function () {
-          if (nodePopoverTimeout) {
-            $timeout.cancel(nodePopoverTimeout);
-          }
+      if (nodePopoverTimeout) {
+        $timeout.cancel(nodePopoverTimeout);
+      }
 
-          nodePopoverTimeout = $timeout(function () {
-            nodeInfo.popover.show();
-          });
+      if (nodeInfo.element && nodeInfo.scope) {
+        nodeInfo.popover = $popover(nodeInfo.element, {
+          trigger: 'manual',
+          placement: 'auto right',
+          target: angular.element(nodeInfo.element[0]),
+          templateUrl: $scope.nodePopoverTemplate,
+          container: 'main',
+          scope: nodeInfo.scope
         });
+        nodeInfo.popover.$promise
+          .then(function () {
+
+            // Needs a timeout here to avoid showing popups instantly when just moving
+            // cursor across a node
```

Comment:
It doesn't work when `trigger` property is set to `manual`, which we're doing here.